### PR TITLE
DOC-12190 active/complete requests API

### PR DESCRIPTION
**Pull request by ggray-cb**
This covers the changes to the Analytics service admin API for active and completed requests.

[Here's a preview of the changes](https://preview.docs-test.couchbase.com/active-complete-requests-api/server/current/analytics/rest-admin.html). 

~~Note: this will need to wait until 7.6.2 is ready to ship before I can merge, because we currently don't have a staging branch in this repo.~~ Thanks to Simon, this will just merge into the staging branch.

### CODE REVIEW

File: docs/modules/analytics/partials/rest-admin/definitions.adoc [hunk 1]
```diff
@@ -288,4 +289,73 @@ __optional__|The number of mutations in the DCP queue that have not yet been ing
...
+|**cancellable** +
+__optional__|Whether this request can be cancelled.|boolean
+|**clientContextID** +
+__optional__|A context ID for debugging purposes. +
+**Example** : `"28379d60-7139-44d6-b57a-95935540b586"`|string
+|**elapsedTime** +
+__optional__|Seconds the request has been running. +
+**Example** : `0.126`|number
+|**jobCreateTime** +
+__optional__|The date and time when the request was created. +
```
Comment (AliSolaiman) [inline docs/modules/analytics/partials/rest-admin/definitions.adoc:309]: the request's job

File: docs/modules/analytics/partials/rest-admin/definitions.adoc [hunk 2]
```diff
@@ -288,4 +289,73 @@ __optional__|The number of mutations in the DCP queue that have not yet been ing
...
+**Example** : `"2024-05-28T19:47:02.512+0000"`|string
+|**jobId** +
+__optional__|The ID of the request. This value can be null if request did not run (for example, due to an error). +
```
Comment (AliSolaiman) [inline docs/modules/analytics/partials/rest-admin/definitions.adoc:312]: The job ID

File: docs/modules/analytics/partials/rest-admin/definitions.adoc [hunk 3]
```diff
@@ -288,4 +289,73 @@ __optional__|The number of mutations in the DCP queue that have not yet been ing
...
+**Example** : `"JID:0.14"`|string
+|**jobQueueTime** +
+__optional__|How long the request has been in the queue waiting to run. +
```
Comment (AliSolaiman) [inline docs/modules/analytics/partials/rest-admin/definitions.adoc:315]: the job has been ...

File: docs/modules/analytics/partials/rest-admin/definitions.adoc [hunk 4]
```diff
@@ -288,4 +289,73 @@ __optional__|The number of mutations in the DCP queue that have not yet been ing
...
+[options="header", cols=".^3a,.^11a,.^4a"]
+|===
+|Name|Description|Schema
+|**cancellable** +
+__optional__|Whether this request can be cancelled.|boolean
+|**clientContextID** +
+__optional__|A context ID for debugging purposes. +
+**Example** : `"28379d60-7139-44d6-b57a-95935540b586"`|string
+|**elapsedTime** +
+__optional__|Seconds the request has been running. +
```
Comment (AliSolaiman) [inline docs/modules/analytics/partials/rest-admin/definitions.adoc:306]: How long the request has been running in seconds.

File: docs/modules/analytics/partials/rest-admin/definitions.adoc [hunk 5]
```diff
@@ -288,4 +289,73 @@ __optional__|The number of mutations in the DCP queue that have not yet been ing
...
+__optional__|The ID of the request. This value can be null if request did not run (for example, due to an error). +
+**Example** : `"JID:0.14"`|string
+|**jobQueueTime** +
+__optional__|How long the request has been in the queue waiting to run. +
+**Example** : `0.0`|number
+|**jobRequiredCPUs** +
+__optional__|The number of CPU cores required to run this request. +
+**Example** : `1`|integer
+|**jobRequiredMemory** +
+__optional__|The bytes of RAM being used to process this request. +
```
Comment (AliSolaiman) [inline docs/modules/analytics/partials/rest-admin/definitions.adoc:321]: The memory required in bytes to run this request.

File: docs/modules/analytics/partials/rest-admin/definitions.adoc [hunk 6]
```diff
@@ -288,4 +289,73 @@ __optional__|The number of mutations in the DCP queue that have not yet been ing
...
+**Example** : `34013184`|integer
+|**jobStartTime** +
+__optional__|The date and time the request began running. +
+**Example** : `"2024-05-28T19:47:02.514+0000"`|string
+|**jobStatus** +
+__optional__|The state of the request. +
```
Comment (AliSolaiman) [inline docs/modules/analytics/partials/rest-admin/definitions.adoc:327]: The state of the job.

File: docs/modules/analytics/partials/rest-admin/definitions.adoc [hunk 7]
```diff
@@ -288,4 +289,73 @@ __optional__|The number of mutations in the DCP queue that have not yet been ing
...
+**Example** : `"RUNNING"`|enum (RUNNING, TERMINATED)
```
Comment (AliSolaiman) [inline docs/modules/analytics/partials/rest-admin/definitions.adoc:328]: (PENDING, RUNNING, TERMINATED, FAILURE, FAILURE_BEFORE_EXECUTION)

File: docs/modules/analytics/partials/rest-admin/definitions.adoc [hunk 8]
```diff
@@ -288,4 +289,73 @@ __optional__|The number of mutations in the DCP queue that have not yet been ing
...
+|**plan** +
+__optional__|The query plan for this request.|string
+|**node** +
+__optional__|The Analytics node running the request. +
```
Comment (AliSolaiman) [inline docs/modules/analytics/partials/rest-admin/definitions.adoc:332]: The Analytics node that received the request.

File: docs/modules/analytics/partials/rest-admin/definitions.adoc [hunk 9]
```diff
@@ -288,4 +289,73 @@ __optional__|The number of mutations in the DCP queue that have not yet been ing
...
+**Example** : `"172.20.0.2:8095"`|string
+|**remoteAddr** +
+__optional__|The network address and port of the system that made the request. +
```
Comment (AliSolaiman) [inline docs/modules/analytics/partials/rest-admin/definitions.adoc:335]: of the client that made the request.

File: docs/modules/analytics/partials/rest-admin/definitions.adoc [hunk 10]
```diff
@@ -288,4 +289,73 @@ __optional__|The number of mutations in the DCP queue that have not yet been ing
...
+**Example** : `"172.20.0.123:53612"`|string
+|**requestTime** +
+__optional__|The date and time the request was made. +
```
Comment (AliSolaiman) [inline docs/modules/analytics/partials/rest-admin/definitions.adoc:338]: the request was received.

File: docs/modules/analytics/partials/rest-admin/definitions.adoc [hunk 11]
```diff
@@ -288,4 +289,73 @@ __optional__|The number of mutations in the DCP queue that have not yet been ing
...
+**Example** : `"2024-05-28T19:44:07.730+0000"`|string
+|**scanConsistency** +
+__optional__|The Scan Consistency setting used for the request's query. +
+**Example** : `"not_bounded"`|string
+|**state** +
+__optional__|The state of the request. +
+**Example** : `"running"`|enum (running, completed)
```
Comment (AliSolaiman) [inline docs/modules/analytics/partials/rest-admin/definitions.adoc:345]: (received, running, cancelled, completed)

File: docs/modules/analytics/partials/rest-admin/definitions.adoc [hunk 12]
```diff
@@ -288,4 +289,73 @@ __optional__|The number of mutations in the DCP queue that have not yet been ing
...
+__optional__|How long the request has been in the queue waiting to run. +
+**Example** : `0.0`|number
+|**jobRequiredCPUs** +
+__optional__|The number of CPU cores required to run this request. +
+**Example** : `1`|integer
+|**jobRequiredMemory** +
+__optional__|The bytes of RAM being used to process this request. +
+**Example** : `34013184`|integer
+|**jobStartTime** +
+__optional__|The date and time the request began running. +
```
Comment (AliSolaiman) [inline docs/modules/analytics/partials/rest-admin/definitions.adoc:324]: the job began running.
OR
the request's job began running.

File: src/analytics-admin/swagger/analytics-admin.yaml
Comment (simon-dew) [inline src/analytics-admin/swagger/analytics-admin.yaml:1]: You've stated that the Active Requests and Completed Requests operations take Analytics Manage / Analytics Select security. For consistency with the other operations, can you add links back from this security definition to these operations? (This isn't an essential part of the spec, I just added it for the convenience of readers.)

So, in the `securityDefinitions.Analytics Manage / Analytics Select.description` key, update the first full line of text in the description to say:

```markdown
For the [Active Requests](#_return_active_requests), [Request Cancellation](#_cancel_request), [Completed Requests](#_completed_requests), [Ingestion Status](#_ingestion_status), and [Pending Mutations](#_monitor_node) operations, users must have one of the following access roles:
```

**Review by simon-dew — CHANGES_REQUESTED** [review]
LGTM, just one change requested.

I need to update the overall [Analytics REST API landing page](https://docs.couchbase.com/server/current/analytics/rest-analytics.html) to fit in with Tony's overall [REST API overview page](https://docs.couchbase.com/server/current/rest-api/rest-intro.html). I'll do this as part of DOC-12204. I'll update the Analytics REST API landing page to include the Active Requests and Completed Requests operations when I do so.

**Comment by simon-dew** [comment]
> Note: this will need to wait until 7.6.2 is ready to ship before I can merge, because we currently don't have a staging branch in this repo

Sorry for this oversight! I have created a `prerelease/7.6.2` branch for this repo and took the liberty to change the base branch for this pull request.

**Comment by simon-dew** [comment]
> I need to update the overall [Analytics REST API landing page](https://docs.couchbase.com/server/current/analytics/rest-analytics.html) to fit in with Tony's overall [REST API overview page](https://docs.couchbase.com/server/current/rest-api/rest-intro.html). I'll do this as part of [DOC-12204](https://issues.couchbase.com/browse/DOC-12204). I'll update the Analytics REST API landing page to include the Active Requests and Completed Requests operations when I do so.

@ggray-cb as promised, I've opened a pull request to update the Analytics REST API landing page to include the Active Requests and Completed Requests operations here:

* https://github.com/couchbase/docs-server/pull/3605

**Comment by ggray-cb** [comment]
OK, I believe I have addressed all the comments, and updated the preview site with the latest.

**Review by simon-dew — APPROVED** [review]
LGTM 👍 

Swagger2Markup is not putting the "Couchbase Server 7.6.2" tag where I wanted it to go, but looking at the [docs](http://swagger2markup.github.io/swagger2markup/1.3.3/#_pathsdocumentextension), seems like that can't be helped 😞

**Pull request merged**